core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -87,6 +89,45 @@ object Invoker {
 
     val msgProvider = SpiLoader.get[MessagingProvider]
     val producer = msgProvider.getProducer(config, ec)
+
+    val proposedInvokerId = if (args.length == 1) { args(0).toInt } else { -1 }
+    val invokerName = sys.env.getOrElse("INVOKER_ASSIGNED_NAME", (() => {
+      if (proposedInvokerId == -1) {
+        logger.error(this, "Must either define INVOKER_ASSIGNED_NAME or provide proposedInvokerId on command line")
+        abort()
+      }
+      UUID().toString
+    })())
```

Comment:
You don't need to use `-1` to indicate "It's not there". Scala has the `Option` type to model just that.

In this particular case, you can make use of some nice helpers to make use of Option's features to fold this nicely:

```
val proposedInvokerId: Option[Int] = args.headOption.map(_.toInt)
val invokerName: String = sys.env.getOrElse("INVOKER_ASSIGNED_NAME", {
  if(proposedInvokerId.isDefined) {
    logger.error(this, "Must either define INVOKER_ASSIGNED_NAME or provide proposedInvokerId on command line")
    abort()
  }
  UUID().toString
})
```

Note: You also don't need to build an anonymous self-calling function here, a block is sufficient. You can now model the sent message to have `Option` as well and handle the absence of a proposal in the controller accordingly.